### ISSUE TITLE
pass mybatis bitmap through the stream to query

### ISSUE THREAD

**Issue by yanye666** [ClickHouse/clickhouse-java#1226]
I want to pass mybatis bitmap through the stream to query, can it be implemented?

![telegram-cloud-document-5-6257914263923329575](https://user-images.githubusercontent.com/42414099/216821881-6577b15a-94db-42cd-aaf2-4e8c5ab9a0ed.jpg)

Like this, can you pass it to bitmap and filter it with another table?

**Comment by zhicwu** [comment]
Yes, via an external or temporary table. Should be similar as: https://github.com/ClickHouse/clickhouse-java/blob/main/examples/jdbc/src/main/java/com/clickhouse/examples/jdbc/Advanced.java#L63-L83

**Comment by yanye666** [comment]
java.sql.SQLException: External table, input function, and query parameter cannot be used together in PreparedStatement.

![image](https://user-images.githubusercontent.com/42414099/218966597-e1384da5-ae06-4a68-839f-e56f044d8ff9.png)

Can't support parameters?

**Comment by yanye666** [comment]
@zhicwu

**Comment by zhicwu** [comment]
> java.sql.SQLException: External table, input function, and query parameter cannot be used together in PreparedStatement.

Yes, you have to pick one but not all of them due to limitation of current implementation. I hope it can be addressed later when I add new SQL parser in `clickhouse-parser`.

Meanwhile, perhaps you can break down the query into two and execute them within a session? For example: first query to create the temp table, and then the second for table join with custom filter/parameter.
